### PR TITLE
SNOW-1457758: Disable telemetry triggering extra describe queries

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1457758

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Short term fix to not trigger duplicate queries when using `session.sql()` to run describe, truncate, copy queries. When collecting telemetry for finding the number of duplicate plan subtrees within the query plan, we end up calling `SelectSQL.to_subqueryable` for the `SelectSQL` plan node generated by `session.sql`. During this step, we call `analyze_attributes` for the given query twice - once during initializing a new subqueryable and second when initializing column states. When these describe queries are run from within a stored sproc, they are being detected as regular queries instead of describe queries and getting executed 3 times. We are removing telemetry calls triggering describe queries as a mitigation and will implement a root cause fix later.

image

### CODE REVIEW

**Review by sfc-gh-jrose — APPROVED** [review]

**Comment by sfc-gh-aalam** [comment]
wait for https://ci-dev-142.int.snowflakecomputing.com/job/PythonStoredProcBuildPrecommitTest/1177/ green before merging

**Review by sfc-gh-yuwang — APPROVED** [review]
LGTM, but please merge after regression succeed

**Review by sfc-gh-yixie — APPROVED** [review]

**Pull request merged**
